sound/soc/sof/intel/tgl.c

```diff
@@ -20,6 +20,54 @@ static const struct snd_sof_debugfs_map tgl_dsp_debugfs[] = {
 	{"dsp", HDA_DSP_BAR,  0, 0x10000, SOF_DEBUGFS_ACCESS_ALWAYS},
 };
 
+static int tgl_dsp_core_get(struct snd_sof_dev *sdev, int core)
+{
+	struct sof_ipc_pm_core_config pm_core_config = {
+		.hdr = {
+			.cmd = SOF_IPC_GLB_PM_MSG | SOF_IPC_PM_CORE_ENABLE,
+			.size = sizeof(pm_core_config),
+		},
+		.enable_mask = sdev->enabled_cores_mask | BIT(core),
+	};
+
+	/* nothing to do if current core ref_count is > 0 */
+	if (sdev->dsp_core_ref_count[core] > 0)
+		return 0;
```

Comment:
should this test/return be moved to the definition of core_get()?

Why repeat the same thing in all platform-specific implementations?